build.py

```diff
@@ -198,6 +193,14 @@ def main():
             parser.exit(1)
 
         # Unpack downloads
+        DIRECTX = source_tree / 'third_party' / 'microsoft_dxheaders' / 'src'
+        ESBUILD = source_tree / 'third_party' / 'devtools-frontend' / 'src' / 'third_party' / 'esbuild'
+        if DIRECTX.exists():
+            shutil.rmtree(DIRECTX)
+            DIRECTX.mkdir()
+        if ESBUILD.exists():
+            shutil.rmtree(ESBUILD)
+            ESBUILD.mkdir()
```

Comment:
Can you please elaborate why we need to purge those folders?

Reply:
Whenever the source tree is reset and updated, `clone.py` will clean the tree of untracked files and directories. It doesn't delete those two specific directories because:
* `microsoft_dxheaders` is a submodule.
* `third_party/devtools-frontend/src/.gitignore` has `third_party/*` on its list.

Because of those reasons, we would have to delete them ourselves. Modifying `.gitignore` isn't an option because `clone.py` runs `git reset` before `git clean`, so any changes we make there would be useless.